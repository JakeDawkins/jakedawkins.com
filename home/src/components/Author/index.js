import React, { Component, PropTypes } from "react";
import { Link } from "react-router";

// XXX remove
import data from "../../api/data.js";

class Author extends Component {

  static propTypes = {
    userDisplayName: PropTypes.string,
    userImageUrl: PropTypes.string,
    userDescription: PropTypes.string,
    userEmail: PropTypes.string,
  }

  render() {
    return(
      <section className="author" itemProp="author">
          <div className="details" itemScope itemType="http://schema.org/Person">
              <img itemProp="image" className="img-rounded" src={data.user.image} alt={data.user.name} />
              <p className="def">Author</p>
              <h3 className="name">
                  <Link itemProp="name" to="/about">{data.user.name}</Link>
              </h3>
              <p className="desc">{data.user.user_description}</p>
              <a itemProp="email" className="email" href="mailto:{data.user.email}">{data.user.email}</a>
          </div>
      </section>
    );
  }
}

export default Author;
