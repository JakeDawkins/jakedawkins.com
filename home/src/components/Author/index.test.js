import Author from './';
import React from 'react';
import renderer from 'react-test-renderer';

describe("Author", () => {

  it("should exist", () => {
    const author = renderer.create(
      <Author />
    );

    expect(author).toBeDefined();
  });

  it("should render props correctly", () => {
    const author = renderer.create(
      <Author
        userDisplayName="Test Name"
        userImageUrl="123456"
        userDescription="this is a description"
        userEmail="1234@1234.com"
      />
    );

    expect(author).toMatchSnapshot();
  });

});
