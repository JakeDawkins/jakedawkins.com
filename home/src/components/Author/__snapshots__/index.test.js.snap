exports[`Author should render props correctly 1`] = `
<section
  className="author"
  itemProp="author">
  <div
    className="details"
    itemScope={true}
    itemType="http://schema.org/Person">
    <img
      alt="Test Name"
      className="img-rounded"
      itemProp="image"
      src="123456" />
    <p
      className="def">
      Author
    </p>
    <h3
      className="name">
      <a
        itemProp="name"
        onClick={[Function bound handleClick]}
        style={Object {}}>
        Test Name
      </a>
    </h3>
    <p
      className="desc">
      this is a description
    </p>
    <a
      className="email"
      href="mailto:1234@1234.com"
      itemProp="email">
      1234@1234.com
    </a>
  </div>
</section>
`;
